### PR TITLE
Add resourceId to expanded messages and doi requests

### PR DESCRIPTION
**Pull request by janbj**

### CODE REVIEW

File: expansion/src/main/java/no/unit/nva/expansion/model/ExpandedMessage.java [hunk 1]
```diff
@@ -27,6 +27,7 @@
     private MessageStatus status;
     private String sender;
     private SortableIdentifier resourceIdentifier;
+    private URI resourceId;
```
Comment (brinxmat) [inline expansion/src/main/java/no/unit/nva/expansion/model/ExpandedMessage.java:30]: Should these not all be "id"?

File: expansion/src/main/java/no/unit/nva/expansion/ResourceExpansionService.java [hunk 2]
```diff
@@ -12,4 +13,6 @@
     ExpandedDataEntry expandEntry(DataEntry dataEntry) throws JsonProcessingException, NotFoundException;
 
     Set<URI> getOrganizationIds(DataEntry dataEntry) throws NotFoundException;
+
+    URI getResourceId(SortableIdentifier identifier);
```
Comment (axthosarouris) [inline expansion/src/main/java/no/unit/nva/expansion/ResourceExpansionService.java:17]: I do not think that this method is necessary. Then we create some weird dependencies where the data entry needs the service and the service needs the data entries.
Suggestion:
```
public static ExpandedDoiRequest create(DoiRequest doiRequest,
                                            ResourceExpansionService resourceExpansionService)
        throws NotFoundException {
        ExpandedDoiRequest expandedDoiRequest = ExpandedDoiRequest.fromDoiRequest(doiRequest);
        Set<URI> ids = resourceExpansionService.getOrganizationIds(doiRequest);
        expandedDoiRequest.setOrganizationIds(ids);
        URI resourceId =  new UriWrapper(ExpansionConstants.ID_NAMESPACE)
            .addChild(expandedDoiRequest.getIdentifier().toString()).getUri();
        expandedDoiRequest.setResourceId(resourceId);
        return expandedDoiRequest;
    }
```

File: expansion/src/main/java/no/unit/nva/expansion/ResourceExpansionServiceImpl.java [hunk 3]
```diff
@@ -60,4 +62,9 @@ public ExpandedDataEntry expandEntry(DataEntry dataEntry) throws JsonProcessingE
         }
         return Collections.emptySet();
     }
+
+    @Override
+    public URI getResourceId(SortableIdentifier identifier) {
+        return new UriWrapper(ExpansionConstants.ID_NAMESPACE).addChild(identifier.toString()).getUri();
```
Comment (axthosarouris) [inline expansion/src/main/java/no/unit/nva/expansion/ResourceExpansionServiceImpl.java:68]: you can put this code directly into the creation of the objects as shown above

File: expansion/src/main/java/no/unit/nva/expansion/model/ExpandedMessage.java (hunk 1 above)
Comment (axthosarouris) [reply expansion/src/main/java/no/unit/nva/expansion/model/ExpandedMessage.java:30]: I think that the misunderstanding is actually the term Resource. We use it in two different ways. One way is as a better name to Publication, but we also view everything in the database as a Resource. e.g. a message is a Resource and a DoiRequest is a Resource. We can agree that all things are Resources, but we should find a name for the Publication that is unique.

**Review by axthosarouris — CHANGES_REQUESTED** [review]

File: expansion/src/main/java/no/unit/nva/expansion/model/ExpandedMessage.java (hunk 1 above)
Comment (janbj) [reply expansion/src/main/java/no/unit/nva/expansion/model/ExpandedMessage.java:33]: Skal vi ta det i https://unit.atlassian.net/browse/NP-3524 ?

**Review by brinxmat — APPROVED** [review]

**Review by axthosarouris — APPROVED** [review]

**Pull request merged**